utils/create_dataset.py

```diff
@@ -31,39 +28,8 @@ def append_to_dataset(dataset, sample):
     return old_size
 
 
-def create_files_and_datasets(samples_size: int, number_of_bands: int, samples_folder: Path, cfg: DictConfig):
-    """
-    Function to create the hdfs files (trn, val and tst).
-    :param samples_size: size of individual hdf5 samples to be created
-    :param number_of_bands: number of bands in imagery
-    :param samples_folder: (str) Path to the output folder.
-    :param cfg: (dict) Parameters found in the yaml config file.
-    :return: (hdf5 datasets) trn, val ant tst datasets.
-    """
-    real_num_bands = number_of_bands
-    assert real_num_bands > 0, "invalid number of bands when accounting for meta layers"
-    hdf5_files = []
-    for subset in ["trn", "val", "tst"]:
-        hdf5_file = h5py.File(os.path.join(samples_folder, f"{subset}_samples.hdf5"), "w")
-        hdf5_file.create_dataset("sat_img", (0, samples_size, samples_size, real_num_bands), np.uint16,
-                                 maxshape=(None, samples_size, samples_size, real_num_bands))
-        hdf5_file.create_dataset("map_img", (0, samples_size, samples_size), np.int16,
-                                 maxshape=(None, samples_size, samples_size))
-        hdf5_file.create_dataset("meta_idx", (0, 1), dtype=np.int16, maxshape=(None, 1))
-        try:
-            hdf5_file.create_dataset("metadata", (0, 1), dtype=h5py.string_dtype(), maxshape=(None, 1))
-            hdf5_file.create_dataset("sample_metadata", (0, 1), dtype=h5py.string_dtype(), maxshape=(None, 1))
-            hdf5_file.create_dataset("params", (0, 1), dtype=h5py.string_dtype(), maxshape=(None, 1))
-            append_to_dataset(hdf5_file["params"], repr(OmegaConf.create(OmegaConf.to_yaml(cfg, resolve=True))))
-        except AttributeError:
-            logging.exception(f'Update h5py to version 2.10 or higher')
-            raise
-        hdf5_files.append(hdf5_file)
-    return hdf5_files
-
-
 class SegmentationDataset(Dataset):
-    """Semantic segmentation dataset based on HDF5 parsing."""
+    """Semantic segmentation dataset based on input csvs listing pairs of imagery and ground truth patches as .tif."""
```

Comment:
csvs?

Reply:
The SegmentationDataset expects as input a csv, not a path to hdf5 anymore: https://github.com/NRCan/geo-deep-learning/blob/develop/utils/create_dataset.py#L91
With current implementation, I'm noticing that the input csv could also be any text file formatted as a csv, but we could encourage the use of the csv by detailling the docs a bit more.